builder/docker/driver_docker.go

```diff
@@ -260,24 +270,42 @@ func (d *DockerDriver) Login(repo, user, pass string) error {
 }
 
 func (d *DockerDriver) Logout(repo string) error {
+	cmd := exec.Command("docker")
+
+	if d.ConfigDir != "" {
+		cmd.Args = append(cmd.Args, "--config", d.ConfigDir)
+	}
+
```

Comment:
Seeing as you have this same logic defined three times, I wonder if we would benefit from defining a function to create the  command. ⚠️ bad name up ahead

```go
func (d *DockerDriver) BuildCommand(args ...string) *exec.Cmd {
cmd := exec.Command("docker")

if d.ConfigDir != "" {
  cmd.Args = append(cmd.Args, "--config", d.ConfigDir)
}
 
cmd.Args = append(cmd.Args, args...)
return cmd
}

/*
example in Logout
cmd := d.BuildCommand("logout", args...)
example in Pull method
cmd := d.BuildCommand("pull", image)
*/